spec/rest/logger.test.js

```diff
@@ -0,0 +1,149 @@
+"use strict";
+
+define(['shared_helper'], function(helper) {
+	var exports = {},
+		logger;
+
+	/* If we are running node, we mimic the loading mechanism used in /nodejs/realtime.js and /nodejs/rest.js
+	 * to get a reference to the logger. We do this because its implementation is not a module.
+	 * 
+	 * If we are in a browser, the logger is already loaded manually via a script tag in nodeunit.html.
+	 */
+	
+	if (isBrowser) {
+		logger = Logger;
+	} else {
+		var fs = require('fs'),
+			path = require('path'),
+			vm = require('vm'),
+			context = vm.createContext({
+				require:require,
+				console:console,
+				process:process,
+				Buffer:Buffer,
+				setTimeout:setTimeout,
+				setInterval:setInterval,
+				clearTimeout:clearTimeout,
+				clearInterval:clearInterval,
+				global:global
+			}),
+			includeScript = function(name) {
+				var filename = path.resolve(__dirname, name);
+				return vm.runInContext(fs.readFileSync(filename, 'utf8'), context, filename);
+			};
+
+		includeScript('../../common/lib/util/logger.js');
+
+		logger = context.Logger;
+	};
+
+	/*
+	 * Check that the logger was instantiated correctly for testing.
+	 */
+	exports.logger_instantiate = function(test) {
+		test.expect(1);
+
+		test.equal(typeof(logger), 'function', 'Logger has been instantiated');
+
+		test.done();
+	};
+
+	/*
+	 * Check that the logger writes to stdout by default with no errors.
+	 */
+	exports.logger_writes_to_stdout = function(test) {
```

Comment:
> This test has nothing to do with stdout, so should be changed to describe what it's actually testing which is console.log

@mattheworiordan Yes this description remained from when I had actually hooked to the stdout stream, because this is what requirement RSC2 is asking. I gave up on this becauseit only works on node and because I don't think it's what we want to test anyway. The logger doesn't know about standard I/O, only console.log.